File: 프로그래머스/2/140107. 점 찍기/점 찍기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

double Distance (long long d, long long x) {
    return sqrt(pow(d, 2) - pow(x, 2));
}

long long solution(int k, int d) {
    long long answer = 0;
    
    int x = 0;
    while (x * k <= d) {
        int maxY = Distance(d, x * k);
        answer += (maxY / k) + 1;
        
        x++;
    }
    
    return answer;
}

File: 프로그래머스/2/140107. 점 찍기/점 찍기.js
function solution(k, d) {
    let answer = 0;
    
    const DistanceY = (d, x) => {
        return Math.sqrt(Math.pow(d, 2) - Math.pow(x, 2));
    }
    
    let i = 0;
    while(i * k <= d) {
        let maxY = DistanceY(d, i * k);
        
        answer += Math.floor(maxY / k) + 1;
        
        i++;
    }
    
    return answer;
}